test/essential/container.jl

```diff
@@ -0,0 +1,50 @@
+@testset "container.jl" begin
+    @model function test()
+        a ~ Normal(0, 1)
+        x ~ Bernoulli(1)
+        b ~ Gamma(2, 3)
+        1 ~ Bernoulli(x / 2)
+        c ~ Beta()
+        0 ~ Bernoulli(x / 2)
+        x
+    end
+
+    @turing_testset "constructor" begin
+      vi = DynamicPPL.VarInfo()
+      sampler = Sampler(PG(10))
+      model = test()
+      trace = AdvancedPS.Trace(model, sampler, vi, AdvancedPS.TracedRNG())
+
+      # Make sure we link the traces
+      @test haskey(trace.model.ctask.task.storage, :__trace)
```

Comment:
Since `Trace` is now: 
```julia
mutable struct Trace{F,R}
    model::F
    rng::R
end
```
we don't need to do it in the constructor. Should be enough to do:
```julia
taskedmodel = AdvancedPS.LibtaskModel(model)
trace = AdvancedPS.Trace(taskedmodel, rng)
addreference!(trace.model.ctask.task, trace)
```

Reply:
Here: 
https://github.com/TuringLang/Turing.jl/blob/55ca37fd511be0e2568ab322cc932775e337e954/src/mcmc/particle_mcmc.jl#L393

or in the main package:
https://github.com/TuringLang/AdvancedPS.jl/blob/955a5628bbce1038db1a66521db9e4a56c9beadb/ext/AdvancedPSLibtaskExt.jl#L136

Reply:
yes, but we do call on `addreference!` when constructing the trace in this case:
https://github.com/TuringLang/Turing.jl/blob/55ca37fd511be0e2568ab322cc932775e337e954/src/mcmc/particle_mcmc.jl#L382-L395

Reply:
Sorry that was mileading, we do use `addreference!` in the constructor we define for `Trace(::Model, ::Sampler, ::VarInfo, ::TracedRNG)`

Reply:
Sorry that was mileading, we do use `addreference!` in the constructor we define for `Trace(::Model, ::Sampler, ::VarInfo, ::TracedRNG)`